### PR TITLE
Design ui

### PR DESCRIPTION
**Pull request by Rand-Hazem**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,7 @@
 .vscode
 
 # MISC
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+# cash
```
Comment (ZakariaTalhami) [inline .gitignore:8]: typo: cache

File: DjangoJenkinsTraining/settings.py [hunk 2]
```diff
@@ -43,7 +43,7 @@
     'django.middleware.security.SecurityMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
-    'django.middleware.csrf.CsrfViewMiddleware',
+    # 'django.middleware.csrf.CsrfViewMiddleware',
```
Comment (ZakariaTalhami) [inline DjangoJenkinsTraining/settings.py:46]: Commented out code

File: build/views.py [hunk 3]
```diff
@@ -27,11 +28,19 @@ def analyze_action(actions, build_obj):
...
             continue
-        if node['_class'] == 'hudson.model.ParametersAction' or node['_class'] == 'com.tikal.jenkins.plugins.multijob.MultiJobParametersAction':
-            build_obj.param = {x['name']: x['value'] for x in node.get('parameters')}
+        print(node['_class'])
+        if node['_class'] == 'hudson.model.ParametersAction' or node[
+            '_class'] == 'com.tikal.jenkins.plugins.multijob.MultiJobParametersAction':
+            build_obj.param = {x['name']: x['value'] for x in node['parameters']}
+            # print("param "+ node['_class'])
+            # print(build_obj.param)
+            # print()
```
Comment (ZakariaTalhami) [inline build/views.py]: commented out code

File: build/views.py [hunk 4]
```diff
@@ -27,11 +28,19 @@ def analyze_action(actions, build_obj):
...
             count += 1
-        elif node['_class'] == 'hudson.triggers.TimerTrigger$TimerTriggerCause' or node['_class'] == 'hudson.model.Cause$UpstreamCause':
-            build_obj.cause = node.get('shortDescription')
+        elif node['_class'] == 'hudson.model.CauseAction':
+            # build_obj.cause = node['shortDescription']
+            # print("cauees " + node['_class'])
+            # print(build_obj.cause)
```
Comment (ZakariaTalhami) [inline build/views.py]: commented out code.

File: build/views.py [hunk 5]
```diff
@@ -27,11 +29,16 @@ def analyze_action(actions, build_obj):
     for node in actions:
         if not node:
             continue
-        if node['_class'] == 'hudson.model.ParametersAction' or node['_class'] == 'com.tikal.jenkins.plugins.multijob.MultiJobParametersAction':
-            build_obj.param = {x['name']: x['value'] for x in node.get('parameters')}
+        print(node['_class'])
```
Comment (ZakariaTalhami) [inline build/views.py:32]: left while debugging?

File: build/views.py [hunk 6]
```diff
@@ -27,11 +29,16 @@ def analyze_action(actions, build_obj):
...
+        if node['_class'] == 'hudson.model.ParametersAction' or node[
+            '_class'] == 'com.tikal.jenkins.plugins.multijob.MultiJobParametersAction':
+            build_obj.param = {x['name']: x['value'] for x in node['parameters']}
             count += 1
-        elif node['_class'] == 'hudson.triggers.TimerTrigger$TimerTriggerCause' or node['_class'] == 'hudson.model.Cause$UpstreamCause':
-            build_obj.cause = node.get('shortDescription')
+        elif node['_class'] == 'hudson.model.CauseAction':
+            print("*******************")
+            print(node['causes'][0]['shortDescription'])
+            print("***********")
```
Comment (ZakariaTalhami) [inline build/views.py:40]: These as well.

**Review by ZakariaTalhami — CHANGES_REQUESTED** [review]
All great, just some minor nitpicking.

**Comment by ZakariaTalhami** [comment]
Can you give some screenshots?

**Comment by Rand-Hazem** [comment]
image

image

image

image

File: build/views.py (hunk 5 above)
Comment (Rand-Hazem) [reply build/views.py:32]: what?

Comment (ZakariaTalhami) [reply build/views.py:32]: I mean that this might need to be cleaned up, unless this was left by intention.

**Pull request closed without merging**